packages/hierarchy-builder/src/hierarchy-builder/internal/operators/grouping/PropertiesGrouping.ts

```diff
@@ -0,0 +1,323 @@
+          {
+            label: translate("grouping.unspecified-label"),
+            propertyGroupingNodeKey: {
+              ...partialPropertyNodeKeyToAdd,
+              type: "property-grouping:value",
+              formattedPropertyValue: "",
+            },
+          },
+          node,
+        );
+        continue;
+      }
+      groupings.ungrouped.push(node);
+      continue;
+    }
+
+    if (currentProperty.ranges) {
+      if (typeof currentProperty.propertyValue === "number") {
+        const propValue = currentProperty.propertyValue;
+        const matchingRange = currentProperty.ranges.find((range) => propValue >= range.fromValue && propValue <= range.toValue);
+
+        if (matchingRange) {
+          const fromValueTypedPrimitive = {
+            type: Number.isInteger(matchingRange.fromValue) ? "Integer" : "Double",
+            value: matchingRange.fromValue,
+          } as TypedPrimitiveValue;
+          const toValueTypedPrimitive = {
+            type: Number.isInteger(matchingRange.toValue) ? "Integer" : "Double",
+            value: matchingRange.toValue,
+          } as TypedPrimitiveValue;
+
+          const rangeLabel = matchingRange.rangeLabel ?? `${await valueFormatter(fromValueTypedPrimitive)} - ${await valueFormatter(toValueTypedPrimitive)}`;
+          addGroupingToMap(
+            groupings.grouped,
+            `${currentProperty.propertyName}:${matchingRange.fromValue} - ${matchingRange.toValue}(${rangeLabel})`,
+            {
+              label: rangeLabel,
+              propertyGroupingNodeKey: {
+                ...partialPropertyNodeKeyToAdd,
+                type: "property-grouping:range",
+                fromValue: matchingRange.fromValue,
+                toValue: matchingRange.toValue,
+              },
+            },
+            node,
+          );
+          continue;
+        }
+      }
+      if (byProperties.createGroupForOutOfRangeValues) {
+        addGroupingToMap(
+          groupings.grouped,
+          `${currentProperty.propertyName}:Other`,
+          {
+            label: translate("grouping.other-label"),
+            propertyGroupingNodeKey: {
+              ...partialPropertyNodeKeyToAdd,
+              type: "property-grouping:other",
+            },
+          },
+          node,
+        );
+        continue;
+      }
+      groupings.ungrouped.push(node);
+      continue;
+    }
+
+    const propertyClass = await getClass(metadata, byProperties.fullClassName);
+    const property = await propertyClass.getProperty(currentProperty.propertyName);
+    if (!property?.isPrimitive()) {
+      groupings.ungrouped.push(node);
+      continue;
+    }
+    const part = {
+      type: property.primitiveType,
+      extendedType: property.extendedTypeName,
+      koqName: (await property.kindOfQuantity)?.fullName,
+      value: currentProperty.propertyValue,
+    } as TypedPrimitiveValue;
+    const formattedValue = await valueFormatter(part);
```

Comment:
Not sure why you called it `part`, but I think we can just avoid creating a variable for it completely.

```suggestion
    const formattedValue = await valueFormatter({
      type: property.primitiveType,
      extendedType: property.extendedTypeName,
      koqName: (await property.kindOfQuantity)?.fullName,
      value: currentProperty.propertyValue,
    });
```